components/Common/CodeBox/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  DocumentDuplicateIcon,
+  CodeBracketIcon,
+} from '@heroicons/react/24/outline';
+import classNames from 'classnames';
+import type { FC, PropsWithChildren } from 'react';
+import { useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import Button from '@/components/Common/Button';
+import Notification from '@/components/Common/Notification';
+import { useCopyToClipboard } from '@/hooks/useCopyToClipboard';
+
+import styles from './index.module.css';
+
+const languageMap: Record<string, string[]> = {
+  JSX: ['jsx'],
+  JSON: ['json'],
+  XML: ['xml'],
+  YML: ['yml', 'yaml'],
+  Bash: ['bash', 'sh', 'shell', 'zsh', 'shellscript'],
+  Shell: ['console', 'shellsession'],
+  TypeScript: ['ts', 'typescript'],
+  JavaScript: ['cjs', 'mjs', 'js', 'javascript'],
+};
+
+const labelsMap = Object.fromEntries(
+  Object.entries(languageMap).flatMap(([label, languages]) =>
+    languages.map(language => [language, label])
+  )
+);
+
+type CodeBoxProps = {
+  language: string;
+  showLineNumbers?: boolean;
+};
+
+const CodeBox: FC<PropsWithChildren<CodeBoxProps>> = props => {
+  const { children, language, showLineNumbers = false } = props;
+
+  const ref = useRef<HTMLPreElement>(null);
+
+  const [copied, copyToClipboard] = useCopyToClipboard();
+
+  const onCopy = () => {
+    if (ref.current !== null && ref.current.textContent !== null) {
+      copyToClipboard(ref.current.textContent);
+    }
+  };
+
+  return (
+    <div className={styles.root}>
+      <div
+        className={classNames(styles.content, {
+          [styles.showLineNumbers]: showLineNumbers,
+        })}
+      >
+        <pre ref={ref}>{children}</pre>
+      </div>
+      <div className={styles.footer}>
+        <span className={styles.language}>
+          {labelsMap[language] ?? language}
+        </span>
+        <Button type="button" className={styles.action} onClick={onCopy}>
+          <DocumentDuplicateIcon className={styles.icon} />
+          <FormattedMessage id="components.common.codebox.copy" />
+        </Button>
+      </div>
+      <Notification open={copied}>
```

Comment:
FYI, we use a hook to dispatch notifications, we don't invoke it directly. Can you use the hook?